### ISSUE TITLE
Empty object when using fromEach

### ISSUE THREAD

**Issue by perpk** [perpk/json-xform#29]
check
https://runkit.com/perpk/61522fc04d501c0009bf3708

expected here: The result shouldn't have any empty objects at all.

**Comment by perpk** [comment]
By fixing this issue another rather annoying case is handled when a nested object is flattened. An example follows to explain the case in detail.
Given the following mapping:
```
const xFormTemplate = {
      fieldset: [
        {
          fromEach: {
            field: 'root',
            flatten: true,
            fieldset: [
              {
                fromEach: {
                  field: 'body',
                  flatten: true,
                  fieldset: [
                    {
                      from: 'fieldOne'
                    },
                    {
                      fromEach: {
                        field: 'branch',
                        fieldset: [
                          {
                            from: 'branchFieldTwo'
                          }
                        ]
                      }
                    }
                  ]
                }
              }
            ]
          }
        }
      ]
    }
```
To map the source:
```
    const source = {
      root: [
        {
          body: [
            {
              fieldOne: 1,
              branch: [
                {
                  branchFieldOne: 10,
                  branchFieldTwo: 22
                }
              ]
            }
          ]
        }
      ]
    }
```
This would yield the result:
```
        {
          fieldOne: 1,
          branchFieldTwo: [22]
        }
```
Which is counter-intuitive, since there is no apparent reason to have the value of `branchFieldTwo` placed in a single-valued array.
After the fix to this issue is introduced the result is much more intuitive and logical, since the value is a literal.
```
        {
          fieldOne: 1,
          branchFieldTwo: 22
        }
```

**Comment by perpk** [comment]
Fixed and merged by #65
